src/index.ts

```diff
@@ -330,33 +331,47 @@ export const plugin: Plugin<[FlexibleContainerOptions?], Root> = (options) => {
 
         // remove the newline "\n" in the beginning, and get the rest of the value
         value = value.slice(1);
+
+        // If the very next line starts with another fence (e.g., "::: tip" or just ":::")
```

Comment:
Very next line will not start with another fence. All fences (also inner ones) will require empty line before.